src/components/common.js

```diff
@@ -118,56 +121,165 @@ export const ButtonOutline = ({ disabled, children, ...props }) => {
+ * @param tabProps Optionally, properties to add to each tab
+ * @param id Optionally, an ID to apply to the wrapper `nav` element so it can be easily found in the
+ *  integration tests. This defaults to 'tab-navigation', but you should change if it this ID would appear
+ *  more than once on a page
+ * @param children Children, which will be appended to teh end of the tab bar
+ * @param props Any additional properties to add to the container menu element
+ */
+export const TabBar = ({
+  activeTab, tabNames, displayNames = {}, refresh = _.noop, getHref,
+  getOnClick = _.noop, label, tabProps = {}, id = 'tab-navigation', children, ...props
+}) => {
   const navTab = (i, currentTab) => {
     const selected = currentTab === activeTab
     const href = getHref(currentTab)
 
-    return h(Clickable, {
-      role: 'tab',
-      'aria-posinset': i + 1,
+    return ({ forwardedRef, onKeyDown }) => span({
+      key: currentTab,
+      role: 'menuitem',
       'aria-setsize': tabNames.length,
-      'aria-selected': selected,
-      style: { ...Style.tabBar.tab, ...(selected ? Style.tabBar.active : {}) },
-      hover: selected ? {} : Style.tabBar.hover,
-      onClick: href === window.location.hash ? refresh : getOnClick(currentTab),
-      href
+      'aria-posinset': i + 1, // The first tab is 1
+      'aria-current': selected ? 'location' : undefined,
+      style: { display: 'flex', minWidth: 140, flexGrow: 0, alignSelf: 'stretch', alignItems: 'center', textAlign: 'center' }
     }, [
-      div({ style: { marginBottom: selected ? -(Style.tabBar.active.borderBottomWidth) : undefined } }, displayNames[currentTab] || currentTab)
+      h(Clickable, {
+        style: { ...Style.tabBar.tab, ...(selected ? Style.tabBar.active : {}) },
+        hover: selected ? {} : Style.tabBar.hover,
+        onClick: href === window.location.hash ? refresh : getOnClick(currentTab),
+        onKeyDown,
+        ref: forwardedRef,
+        href
+      }, [
+        div({ style: { flex: '1 1 100%', marginBottom: selected ? -(Style.tabBar.active.borderBottomWidth) : undefined } }, displayNames[currentTab] || currentTab)
+      ])
     ])
   }
 
   return div({
-    role: 'tablist',
-    'aria-label': 'Tab bar',
-    style: Style.tabBar.container
+    role: 'navigation',
+    id
   }, [
-    ..._.map(([i, name]) => navTab(i, name), Utils.toIndexPairs(tabNames)),
-    div({ style: { flexGrow: 1 } }),
-    children
+    div({
+      role: 'menu',
+      'aria-label': label,
+      'aria-orientation': 'horizontal',
+      style: Style.tabBar.container,
+      ...props
+    }, [
+      withHorizontalNavigation(_.map(([i, name]) => navTab(i, name), Utils.toIndexPairs(tabNames))),
+      div({ style: { flexGrow: 1 } }),
+      children
+    ])
   ])
 }
+TabBar.propTypes = {
+  activeTab: PropTypes.string.isRequired,
+  tabNames: PropTypes.arrayOf(PropTypes.string).isRequired,
+  displayNames: PropTypes.arrayOf(PropTypes.string),
+  label: PropTypes.string.isRequired,
```

Comment:
For requiring this sort of thing, you might want to look at our uses of `Utils.useConsoleAssert`, which we currently use specifically for ensuring that accessibility labels get added, since sometimes people don't develop with the console open, and it makes things explode